packages/react-instantsearch-core/src/core/__tests__/createInstantSearchManager.derived.js

```diff
@@ -1,340 +1,392 @@
+      props: {},
+    });
 
-        ism.widgetsManager.registerWidget({
-          getSearchParameters: params => params.setIndex('second'),
-          context: { multiIndexContext: { targetedIndex: 'second' } },
-          props: { indexName: 'second' },
-        });
+    expect(ism.store.getState().results).toBe(null);
+
+    return Promise.resolve()
+      .then(() => {})
+      .then(() => {
+        const store = ism.store.getState();
+
+        expect(store.results.first).toEqual(
+          expect.objectContaining({
+            query: 'first query 1',
+            index: 'first',
+            page: 3,
+          })
+        );
 
-        expect(ism.store.getState().results).toBe(null);
-
-        return Promise.resolve()
-          .then(() => {}) // We need to wait for the next tick
-          .then(() => {
-            const store = ism.store.getState();
-            expect(store.results.first).toEqual({
-              query: 'first query 1',
-              page: 3,
-              index: 'first',
-            });
-            expect(store.results.second).toEqual({
-              query: 'second query 1',
-              page: 0,
-              index: 'second',
-            });
-            expect(store.error).toBe(null);
-
-            ism.widgetsManager.getWidgets()[0].getSearchParameters = params =>
-              params.setQuery('first query 2');
-            ism.widgetsManager.getWidgets()[1].getSearchParameters = params =>
-              params.setQuery('second query 2');
-
-            ism.widgetsManager.update();
-
-            return Promise.resolve()
-              .then(() => {})
-              .then(() => {
-                const store1 = ism.store.getState();
-                expect(store.results.first).toEqual({
-                  query: 'first query 2',
-                  page: 3,
-                  index: 'first',
-                });
-                expect(store.results.second).toEqual({
-                  query: 'second query 2',
-                  page: 0,
-                  index: 'second',
-                });
-                expect(store1.error).toBe(null);
-              });
-          });
+        expect(store.results.second).toEqual(
+          expect.objectContaining({
+            query: 'second query 1',
+            index: 'second',
+            page: 0,
+          })
+        );
+      })
+      .then(() => {
+        // <SearchBox defaultRefinement="first query 2" />
+        ism.widgetsManager.getWidgets()[0].getSearchParameters = params =>
+          params.setQuery('first query 2');
+
+        // <Index indexName="second">
+        //   <SearchBox defaultRefinement="second query 2" />
+        // </Index>
+        ism.widgetsManager.getWidgets()[4].getSearchParameters = params =>
+          params.setQuery('second query 2');
+
+        // Simualte an udpate (see `createConnector`)
+        ism.widgetsManager.update();
+      })
+      .then(() => {})
```

Comment:
since I see this twice, I suppose it's not a typo.
Are you trying to skip a tick?

Reply:
I think I've mostly answer to the question inside [the previous comment](https://github.com/algolia/react-instantsearch/pull/1824#discussion_r243545151)